src/transports/smart.c

```diff
@@ -226,6 +226,9 @@ static int git_smart__connect(
 	t->url = git__strdup(url);
 	GIT_ERROR_CHECK_ALLOC(t->url);
 
+	if (t->proxy.url)
```

Comment:
optional: the next line can be called unconditionally: `git_proxy_options_clear` takes care to reset the pointer to `NULL`, so it can be called multiple times without problems.

Reply:
It's correct, I fix it.